cosima_cookbook/database.py

```diff
@@ -43,18 +49,39 @@ class NCExperiment(Base):
     root_dir = Column(String, nullable=False)
 
     # Other experiment metadata (populated from metadata.yaml)
-    metadata_keys = ['contact', 'email', 'created', 'description', 'notes']
+    metadata_keys = ['contact', 'email', 'created', 'description', 'notes', 'keywords']
     contact = Column(String)
     email = Column(String)
     created = Column(DateTime)
     #: Human-readable experiment description
     description = Column(Text)
     #: Any other notes
     notes = Column(Text)
+    #: Short, categorical keywords
+    kw = relationship(
+        'Keyword',
+        secondary=keyword_assoc_table,
+        back_populates='experiments',
+        collection_class=set
+    )
+    # add an association proxy to the keyword column of the keywords table
+    # this lets us add keywords as strings rather than Keyword objects
+    keywords = association_proxy('kw', 'keyword')
 
     #: Files in this experiment
     ncfiles = relationship('NCFile', back_populates='experiment')
 
+class Keyword(Base):
+    __tablename__ = 'keywords'
+
+    id = Column(Integer, primary_key=True)
+    keyword = Column(String, nullable=False, unique=True, index=True)
```

Comment:
I *think* we want the `unique` constraint here, but if we add a new experiment with a keyword that exists already, we get an integrity error :roll_eyes: